cmd/gateway/commands.go

```diff
@@ -283,6 +287,14 @@ func createStaticModeCommand() *cobra.Command {
 		"Use NGINX Plus",
 	)
 
+	cmd.Flags().BoolVar(
```

Comment:
So, that would mean if the user doesn't set this flag to true and the `experimental` annotation is found on the core Gateway API CRDs, we would set `Accepted:false` on the GatewayClass?

That's intriguing. The spec doesn't mention anything about checking the channel annotation as part of the SupportedVersion check:
> The version of Gateway API CRDs that is installed in a cluster can be determined
by looking at the `gateway.networking.k8s.io/bundle-version` annotation on each
CRD. Each implementation MUST compare that with the list of versions that it
recognizes and supports. Implementations with a GatewayClass MUST publish the
`SupportedVersion` condition on the GatewayClass to indicate whether the CRDs
installed in the cluster are supported.

However, reporting an error in the status for this edge case could be beneficial to the user. It would be a different condition and reason than SupportedVersion/UnsupportedVersion since we are dealing with release channels and not versions.

It's worth asking this question in the Gateway API slack channel. 

If we decide to implement this, I don't think it should hold up this PR. We can tackle it in a different PR.

Reply:
> Should we not accept this though? There could be a valid reason for wanting the experimental APIs installed on the cluster but not use the supported features of same in NGF

We could accept it but add a warning, similar to how we handle supported versions where the major version is supported, but the minor isn't. In that case, we set `Accepted: true` and `SupportedVersion: false`.

I think a warning would be helpful because there are likely features in the experimental CRDs that we don't support and that may not be clear to the user without a warning.

Reply:
That makes sense. I'll ask in the sigs channel and open a ticket to follow up on this depending on what they say!